graphdatascience/session/session_sizes.py

```diff
@@ -1,26 +1,53 @@
+from dataclasses import dataclass
 from enum import Enum
 from typing import List
 
 
+@dataclass(frozen=True)
+class SessionMemoryValue:
+    value: str
+
+    def __str__(self) -> str:
+        return self.value
+
+    @staticmethod
+    def fromApiResponse(value: str) -> "SessionMemoryValue":
+        """
+        Converts the string value from an API response to a SessionMemory enumeration value.
+
+        Args:
+            value: The string value from the API response.
+
+        Returns:
+            The SessionMemory enumeration value.
+
+        """
+        if value == "":
+            raise ValueError("memory configuration cannot be empty")
+
+        return SessionMemoryValue(value.replace("Gi", "GB"))
+
+
 class SessionMemory(Enum):
     """
     Enumeration representing session main memory configurations.
     """
 
-    m_8GB = "8GB"
-    m_16GB = "16GB"
-    m_24GB = "24GB"
-    m_32GB = "32GB"
-    m_48GB = "48GB"
-    m_64GB = "64GB"
-    m_96GB = "96GB"
-    m_128GB = "128GB"
-    m_192GB = "192GB"
-    m_256GB = "256GB"
-    m_384GB = "384GB"
+    m_4GB = SessionMemoryValue("4GB")
```

Comment:
as this type is user-facing, we should put it into the `__init__.py` of this package